eng/testing/linker/SupportFiles/Directory.Build.props

```diff
@@ -13,5 +12,8 @@
     <EnableTrimAnalyzer>false</EnableTrimAnalyzer>
     <SuppressTrimAnalysisWarnings>true</SuppressTrimAnalysisWarnings>
     <NoWarn>$(NoWarn);IL2121</NoWarn>
+
+    <!-- EventSourceSupport is disabled by default with PublishAot=true, so need to enable it here. -->
+    <EventSourceSupport>true</EventSourceSupport>
```

Comment:
@MichalStrehovsky @jkotas - with this change we now have tests ensuring EventSource works (at least in proc) with PublishAot. Do you think in .NET 8 we should re-enable `EventSourceSupport=true` by default? Or do you think we should keep it disabled by default going forward (for size savings?)

Reply:
would wait for feedback on the current .NET 7 default before doing more changes.

I think it makes sense to leave it off by default and turn it on only for server app-models in their SDKs.

Reply:
Agree with Jan. I kind of like the size savings. I would like to see feedback. Re-enabling it is an instant 30% size regression from 7.0 for hello world.

Reply:
FYI - @noahfalk @tommcdon

Reply:
Just curious, how big (in bytes) is 30%? I wouldn't have guessed the savings were such a high percentage but I don't know if that is because I overestimated the size of HelloWorld or underestimated the size of EventSource.

Reply:
I was off a little bit. It's 2.85 MB to 3.48 MB, so 22%.

Reply:
Interesting, so 650KB is eliminated just by removing EventSource, certainly more than I expected. I'm not planning to look into now, but I am guessing in the future there is some tooling or log that would help us break down the contribution by type or method to better understand the size? Thanks @MichalStrehovsky!

Reply:
Yes, we have a log and I'm hoping to do some offcycle work to visualize/diff them.

Reply:
I've love to see the results of this.  Agreed with @noahfalk that this is more than I would have expected.  It feels important to describe the diagnostic expectations for native AOT in .NET 8 so that we can determine how this decision fits in.

Reply:
There's a similar difference for `PublishTrimmed` as well - the size of managed code goes from 2,718,848 bytes to 3,003,008. So the cost is about 300 kB in IL terms. (The repro steps for both is basically just run `dotnet publish -r win-x64 -c Release` with `-p:PublishTrimmed=true` for trimming or `-p:PublishAot=true` for NativeAOT. Use `-p:EventSourceSupport=false` to enable/disable EventSource.)

For the NativeAOT case a quick breakdown is following.

# Size of code per namespace with EventSource enabled

Namespace | Size of code in bytes
-- | --
System | 414,414
System.Collections.Generic | 246,832
System.Globalization | 128,834
Internal.Runtime.TypeLoader | 128,583
System.Diagnostics.Tracing | 116,975
System.Text | 92,958
Internal.TypeSystem | 85,341
System.Reflection | 40,905
System.Reflection.Runtime.TypeInfos | 36,802
System.Collections.Concurrent | 34,363
System.Reflection.Runtime.General | 31,092
System.Numerics | 30,384
System.Threading | 27,273
Internal.Reflection.Execution | 27,180
System.IO | 26,659
System.Resources | 25,769
System.Reflection.Runtime.MethodInfos | 22,513
System.Runtime | 21,400
System.Runtime.CompilerServices | 20,287
System.Buffers | 20,055

# Size of code per namespace with EventSource disabled

Namespace | Size of code in bytes
-- | --
System | 306,356
System.Collections.Generic | 156,429
Internal.Runtime.TypeLoader | 127,630
System.Globalization | 107,047
System.Text | 86,065
Internal.TypeSystem | 85,341
System.Reflection | 36,931
System.Reflection.Runtime.TypeInfos | 35,576
System.Collections.Concurrent | 34,363
System.Numerics | 30,384
System.Reflection.Runtime.General | 27,776
System.IO | 26,299
Internal.Reflection.Execution | 25,920
System.Resources | 25,445
System.Runtime | 21,336


(This is just namespaces with more than 20 kB of code)

Most of the cost is in transitive dependencies. EventSource has a big transitive closure and depends on a lot of framework code.

I can file an issue if there's interest in reducing the transtive closure.

Reply:
> Most of the cost is in transitive dependencies. EventSource has a big transitive closure and depends on a lot of framework code.

This tells me that this is the upper bound of the size savings for having EventSource disabled. In a real world app that uses these transitive dependencies, they won't be able to be trimmed. So the size savings of disabling EventSource will be less. For example, 90 kB of size savings is coming from `System.Collections.Generic` which many apps will use. When the app uses those collections, they won't be able to be trimmed anymore, even with EventSource disabled.

Reply:
> So the size savings of disabling EventSource will be less. For example, 90 kB of size savings is coming from System.Collections.Generic which many apps will use. When the app uses those collections, they won't be able to be trimmed anymore, even with EventSource disabled

That depends - when thinking about native code, one needs to think about instantiated generic code. For example the costs associated with `System.Collections.Generic.Dictionary`2<System.Type,System.Diagnostics.Tracing.TraceLoggingTypeInfo>` are exclusive to EventSource not matter what the app does.

It's easy to test out theories - just pick a bigger app and see the impact of EventSource. For apps the size of current ASP.NET, an extra 600 kB overhead is negligible either way,